kube/src/api/streaming.rs

```diff
@@ -0,0 +1,274 @@
+// Code to support streaming resources (attach/exec/portforward)
```

Comment:
Personally, I think the overloaded meaning of streams in rust (streams being a feature of reqwest we use, watch calls returning a Stream, kube-runtime being the abstraction on top of these streaming calls), would make another use of stream/streaming confusing.

Reply:
Good point. I'll rename it just before merging to something else to preserve review comments. (Or is GitHub smart enough to reattach?)

Maybe `remote_command::AttachedProcess` because `attach` and `exec` are in `streaming/remotecommand`.

Any suggestions?

Reply:
`remote_command` probably works. I was thinking we ought to have something that also captures `port-forward` and `proxy` as well, but struggling to come up with something other than just `specialized`.

> Or is GitHub smart enough

I am continually perplexed by how PR review flows reorders things, but it doesn't seem to lose things at least. Push and we'll see :-)